aries_cloudagent/resolver/default/peer3.py

```diff
@@ -0,0 +1,142 @@
+            # retrieve did_doc from storage using did:peer:3
+            async with profile.session() as session:
+                storage = session.inject(BaseStorage)
+                record = await storage.find_record(
+                    RECORD_TYPE_DID_DOCUMENT, {"did": did}
+                )
+                did_doc = DIDDocument.from_json(record.value)
+        else:
+            raise DIDNotFound(f"did is not a did:peer:3 {did}")
+
+        return did_doc.dict()
+
+    async def create_and_store_document(
+        self, profile: Profile, peer_did_2_doc: DIDDocument
+    ):
+        """Injest did:peer:2 document create did:peer:3 and store document."""
+        if not peer_did_2_doc.id.startswith("did:peer:2"):
+            raise MalformedPeerDIDError("did:peer:2 expected")
+
+        dp3_doc = deepcopy(peer_did_2_doc)
+        _convert_to_did_peer_3_document(dp3_doc)
+        try:
+            async with profile.session() as session:
+                storage = session.inject(BaseStorage)
+                record = await storage.find_record(
+                    RECORD_TYPE_DID_DOCUMENT, {"did": dp3_doc.id}
+                )
+        except StorageNotFoundError:
+            record = StorageRecord(
+                RECORD_TYPE_DID_DOCUMENT,
+                dp3_doc.to_json(),
+                {"did": dp3_doc.id},
+            )
+            async with profile.session() as session:
+                storage: BaseStorage = session.inject(BaseStorage)
+                await storage.add_record(record)
+        else:
+            async with profile.session() as session:
+                storage: BaseStorage = session.inject(BaseStorage)
+                await storage.update_record(
+                    record, dp3_doc.to_json(), {"did": dp3_doc.id}
+                )
+        await _reset_keys_from_did_doc(profile, dp3_doc)
+
+        return dp3_doc
+
+
+async def _reset_keys_from_did_doc(profile, did_doc):
+    conn_mgr = BaseConnectionManager(profile)
+    async with profile.session() as session:
+        storage: BaseStorage = session.inject(BaseStorage)
+        await storage.delete_all_records(
+            BaseConnectionManager.RECORD_TYPE_DID_KEY, {"did": did_doc.id}
+        )
+
+    for vm in did_doc.verification_method or []:
+        if vm.controller == did_doc.id:
+            if vm.public_key_base58:
+                await conn_mgr.add_key_for_did(did_doc.id, vm.public_key_base58)
+            if vm.public_key_multibase:
+                pk = multibase.decode(vm.public_key_multibase)
+                if len(pk) == 32:  # No multicodec prefix
+                    pk = bytes_to_b58(pk)
+                else:
+                    codec, key = multicodec.unwrap(pk)
+                    if codec == multicodec.multicodec("ed25519-pub"):
+                        pk = bytes_to_b58(key)
+                    else:
+                        continue
+                await conn_mgr.add_key_for_did(did_doc.id, pk)
```

Comment:
Similar to my other comment, would really like to see at least a docstring on this one to at least describe it's purpose and why it's needed.